### ISSUE TITLE
Broken TriString handling

### ISSUE THREAD

**Issue by mmsteven** [JuliaEditorSupport/julia-vim#99]
Tristring handling is broken for me.  If I scroll through a file or return focus to it then the section of code before/after the tri string gets identified as a the text portion (for syntax highlighting and formatting).

I tried a variety of things to fix it, but none of them worked, and some (such as `:syn sync fromstart` would actually break the formatting if I was in the middle of the file.  The only thing that worked was to actually scroll to the top, then scroll down to the end of the file.

I have this problem with neovim (0.1.5) and vim (7.4.1818). Julia is verion 0.5.0

I spent some time digging around in the syntax file, and found that the offending line is in [syntax/julia.vim:221](https://github.com/JuliaEditorSupport/julia-vim/blob/master/syntax/julia.vim#L221) 

```vim
exec 'syntax region  juliaMacroCall	contained transparent start="\(@' . s:idregex . '\%(\.' . s:idregex . '\)*\)\@=\1\%([^(]\|$\)" end="\ze\%([])};#]\|$\)" contains=@juliaExpressions,juliaMacro,juliaSymbolS,juliaQuotedParBlockS,juliaQuotedQMarkParS'
```

I was not able to fix it myself, but if I comment out that line the problem goes away.

edit: added julia version

**Comment by carlobaldassi** [comment]
I'm not able to reproduce this problem. I also don't see how the juliaMacroCall line may be involved. Do you have an example file (perhaps shared privately if you don't want to post it) and a procedure that I can follow to reproduce the issue? I occasionally experience glitches, and it would be very nice to fix them, but so far they have been very inconsistent/unpredictable.

(By the way: did you try CTRL-L to refresh? With recent versions of the plugin, I find it generally fixes things when syntax occasionally fails)

**Comment by mmsteven** [comment]
@carlobaldassi Thanks for the reply. 

After some more trouble shooting I am now able to reliably replicate the problem.  

To replicate, find a rather large file with tri string comments.

add
```julia
    println("$(@__FILE__)")
```

somewhere in the file.

then `G` to the bottom of the file (don't scroll)

The syntax highlighting should be broken.  If it is not.. then do:
```vim
:syn sync fromstart
```
and it will break.

For example, I was able to replicate the problem with [this public file here](https://raw.githubusercontent.com/JuliaStats/DataFramesMeta.jl/master/src/DataFramesMeta.jl) by adding the ```println("$(@__FILE__")``` statement  (fwiw, the last test I did I added the print statement at line 139, though it doesn't seem to matter at all)
